src/command/command.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import EmitterMixin from '../emittermixin.js';
+import utils from '../utils.js';
+
+/**
+ * The base class for CKEditor commands.
+ *
+ * Commands are main way to manipulate editor contents and state. They are mostly used by UI elements (or by other
+ * commands) to make changes in Tree Model. Commands are available in every part of code that has access to
+ * {@link core.Editor} instance, since they are registered in it and executed through {@link core.Editor#execute}.
+ * Commands instances are available through {@link core.Editor#commands}.
+ *
+ * This is an abstract base class for all commands.
+ *
+ * @memberOf core.command
+ */
+export default class Command {
+	/**
+	 * Creates a new Command instance.
+	 *
+	 * @param {core.Editor} editor Editor on which this command will be used.
+	 */
+	constructor( editor ) {
+		/**
+		 * Editor on which this command will be used.
+		 *
+		 * @member {core.Editor} core.command.Command#editor
+		 */
+		this.editor = editor;
+
+		/**
+		 * Flag indicating whether a command is enabled or disabled.
+		 * A disabled command should do nothing upon it's execution.
+		 *
+		 * @member {Boolean} core.command.Command#isEnabled
+		 */
+		this.isEnabled = true;
+
+		// If schema checking function is specified, add it to the `refreshState` listeners.
+		// Feature will be disabled if it does not apply to schema requirements.
+		if ( this.checkEnabled ) {
+			this.on( 'refreshState', ( evt ) => {
+				if ( !this.checkEnabled() ) {
+					return disableCallback( evt );
+				}
+			} );
+		}
+	}
+
+	/**
+	 * Checks if a command should be enabled according to its own rules. Mostly it will check schema to see if the command
+	 * is allowed to be executed in given position. This method can be defined in child class (but is not obligatory).
+	 * If it is defined, it will be added as a callback to `refreshState` event.
+	 *
+	 * @method core.command.Command#checkEnabled
```

Comment:
Sounds like a protected method, rather than a public one. The public interface is made of a `refreshState()` method and `isEnabled` property.